features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Listing/ApiTagThumb.jsx

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+/**
+ *
+ *
+ * @param {*} theme
+ */
+const styles = theme => ({
+    thumbContent: {
+        width: theme.custom.tagThumbnail.width - theme.spacing.unit,
+        backgroundColor: theme.palette.background.paper,
+        padding: theme.spacing.unit,
+    },
+    thumbLeft: {
+        alignSelf: 'flex-start',
+        flex: 1,
+    },
+    thumbRight: {
+        alignSelf: 'flex-end',
+    },
+    thumbInfo: {
+        display: 'flex',
+    },
+    thumbHeader: {
+        width: theme.custom.tagThumbnail.width - theme.spacing.unit,
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        cursor: 'pointer',
+        justifyContent: 'center',
+        margin: 0,
+    },
+    contextBox: {
+        // eslint-disable-next-line radix
+        width: parseInt((150 - theme.spacing.unit) / 2),
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        cursor: 'pointer',
+        margin: 0,
+        display: 'inline-block',
+        lineHeight: '1em',
+    },
+    thumbWrapper: {
+        position: 'relative',
+        paddingTop: 20,
+        marginRight: theme.spacing.unit * 2,
+    },
+    deleteIcon: {
+        fill: 'red',
+    },
+    textWrapper: {
+        color: theme.palette.text.secondary,
+        textDecoration: 'none',
+    },
+    image: {
+        width: theme.custom.tagThumbnail.width,
+    },
+    imageWrapper: {
+        color: theme.palette.text.secondary,
+        backgroundColor: theme.palette.background.paper,
+        width: theme.custom.tagThumbnail.width + theme.spacing.unit,
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    imageOverlap: {
+        position: 'absolute',
+        bottom: 1,
+        backgroundColor: theme.custom.thumbnail.contentBackgroundColor,
+    },
+});
+/**
+ *
+ *
+ * @class ApiTagThumb
+ * @extends {React.Component}
+ */
+
+
+class ApiTagThumb extends React.Component {
```

Comment:
Shall we make this a functional component

Reply:
Fixed c62fedce0fd3452f947ee5d7b2a33c0f3dc5959f